generic-contracts/src/integration-test/java/com/scalar/dl/genericcontracts/GenericContractTableEndToEndTest.java

```diff
@@ -0,0 +1,1569 @@
+    select.set(Constants.QUERY_JOINS, createArrayNode(join));
+
+    // Act
+    ContractExecutionResult actual = clientService.executeContract(CONTRACT_ID_SELECT, select);
+
+    // Assert
+    assertThat(actual.getContractResult()).isPresent();
+    assertSelectResult(
+        jacksonSerDe.deserialize(actual.getContractResult().get()), ImmutableList.of());
+  }
+
+  @Test
+  public void
+      select_RecordsWithJsonObjectColumnAndNumericTypeConditionGiven_ShouldNotReturnRecords() {
+    // Arrange
+    JsonNode record1 =
+        prepareRecord(
+            ImmutableMap.of(
+                COLUMN_NAME_1,
+                TextNode.valueOf("aaa"),
+                COLUMN_NAME_2,
+                IntNode.valueOf(1),
+                COLUMN_NAME_3,
+                mapper.createObjectNode()));
+    JsonNode record2 =
+        prepareRecord(
+            ImmutableMap.of(
+                COLUMN_NAME_1,
+                TextNode.valueOf("bbb"),
+                COLUMN_NAME_2,
+                IntNode.valueOf(1),
+                COLUMN_NAME_3,
+                mapper.createArrayNode()));
+    clientService.executeContract(CONTRACT_ID_INSERT, prepareInsert(COMMON_TEST_TABLE, record1));
+    clientService.executeContract(CONTRACT_ID_INSERT, prepareInsert(COMMON_TEST_TABLE, record2));
+    JsonNode select =
+        prepareSelect(
+            COMMON_TEST_TABLE,
+            COLUMN_NAME_2,
+            IntNode.valueOf(1),
+            prepareCondition(COLUMN_NAME_3, IntNode.valueOf(10), Constants.OPERATOR_NE));
+
+    // Act
+    ContractExecutionResult actual = clientService.executeContract(CONTRACT_ID_SELECT, select);
+
+    // Assert
+    assertThat(actual.getContractResult()).isPresent();
+    assertSelectResult(
+        jacksonSerDe.deserialize(actual.getContractResult().get()), ImmutableList.of());
+  }
+
+  @Test
+  public void select_JsonObjectTypeConditionsGiven_ShouldThrowContractContextException() {
+    // Arrange
+    JsonNode condition1 =
+        prepareCondition(COLUMN_NAME_3, mapper.createObjectNode(), Constants.OPERATOR_EQ);
+    JsonNode condition2 =
+        prepareCondition(COLUMN_NAME_3, mapper.createArrayNode(), Constants.OPERATOR_EQ);
+    JsonNode select1 =
+        prepareSelect(COMMON_TEST_TABLE, COLUMN_NAME_2, IntNode.valueOf(1), condition1);
+    JsonNode select2 =
+        prepareSelect(COMMON_TEST_TABLE, COLUMN_NAME_2, IntNode.valueOf(1), condition2);
+
+    // Act Assert
+    assertThatThrownBy(() -> clientService.executeContract(CONTRACT_ID_SELECT, select1))
+        .isExactlyInstanceOf(ClientException.class)
+        .hasMessage(Constants.INVALID_CONDITION_FORMAT + condition1);
+    assertThatThrownBy(() -> clientService.executeContract(CONTRACT_ID_SELECT, select2))
+        .isExactlyInstanceOf(ClientException.class)
+        .hasMessage(Constants.INVALID_CONDITION_FORMAT + condition2);
+  }
+
+  @Test
+  public void select_EqualButDifferentNumericTypePrimaryKeysGiven_ShouldReturnRecordProperly() {
```

Comment:
Several tests from here are the corresponding tests for the numeric value handling fix.